fluent.runtime/tests/utils.py

```diff
@@ -1,5 +1,58 @@
+"""Utilities for testing."""
+
 import textwrap
+from pathlib import PurePath
+from unittest import mock
+from io import StringIO
+import functools
 
 
 def dedent_ftl(text):
     return textwrap.dedent(f"{text.rstrip()}\n")
+
+
+# Unify path separator, default path separator on Windows is \ not /
+# Supports only relative paths
+# Needed in test_falllback.py because it uses dict + string compare to make a virtual file structure
+def _normalize_path(path):
+    path = PurePath(path)
+    if path.is_absolute():
+        raise ValueError("Absolute paths are not supported in file simulation yet. ("
+                         + str(path) + ")")
+    if "." not in path.parts and ".." not in path.parts:
+        return "/".join(PurePath(path).parts)
+    else:
+        res_parts = []
+        length = len(path.parts)
+        i = 0
+        while i < length:
+            if path.parts[i] == ".":
+                i += 1
+            elif i < length - 1 and path.parts[i+1] == "..":
+                i += 2
+            else:
+                res_parts.append(path.parts[i])
+                i += 1
+        return "/".join(res_parts)
```

Comment:
This seems unnecessarily complicated. Couldn't we just not support `.` or `..` within paths? That seems like it would be simpler.

```suggestion
    if path.is_absolute() or "." in path.parts or ".." in path.parts:
        raise ValueError(f"Unsupported path: {path}")
    return "/".join(path.parts)
```